sandbox/apparmor/notify/listener/listener.go

```diff
@@ -323,18 +298,20 @@ func (l *Listener) Run() error {
 	// is called) return and close the tomb's dead channel, as it is the last
 	// and only tracked goroutine.
 	l.tomb.Go(func() error {
-		var err error
 		for {
-			err = l.tomb.Err()
-			if err != tomb.ErrStillAlive {
+			if l.tomb.Err() != tomb.ErrStillAlive {
 				break
 			}
-			err = l.runOnce()
+			err := l.runOnce()
 			if err != nil {
-				break
+				if exitOnError {
+					return err
+				} else {
+					logger.Noticef("error in listener run loop: %v", err)
```

Comment:
can the code really continue at this point?

Reply:
Yes, that was the idea behind logging the error instead of returning it. Since all the other internal errors are now logged instead of returned from within tomb-tracked goroutines, the tomb is only ever killed by `listener.Close()`, so the listener will happy carry on even if an error occurs.

Reply:
s/listener/prompting listener/ ?